webapi/src/main/scala/org/knora/webapi/responders/v1/UsersResponderV1.scala

```diff
@@ -215,20 +210,20 @@ final case class UsersResponderV1Live(
     userProfile: UserProfileV1
   ): Task[UserProfileResponseV1] =
     for {
-      _ <- ZIO.attempt(
-             if (!userProfile.permissionData.isSystemAdmin && !userProfile.userData.user_id.contains(userIRI)) {
-               throw ForbiddenException("SystemAdmin permissions are required.")
-             }
-           )
+      _ <-
+        ZIO
+          .fail(ForbiddenException("SystemAdmin permissions are required."))
+          .when(!userProfile.permissionData.isSystemAdmin && !userProfile.userData.user_id.contains(userIRI))
+
       maybeUserProfileToReturn <- userProfileByIRIGetV1(
                                     userIri = userIRI,
                                     profileType = profileType
                                   )
 
-      result = maybeUserProfileToReturn match {
-                 case Some(up) => UserProfileResponseV1(up)
-                 case None     => throw NotFoundException(s"User '$userIRI' not found")
-               }
+      result <- maybeUserProfileToReturn match {
+                  case Some(up) => ZIO.succeed(UserProfileResponseV1(up))
+                  case None     => ZIO.fail(NotFoundException(s"User '$userIRI' not found"))
+                }
```

Comment:
this could be expresses with ZIO.fromOption.

And in fact the above `userProfileByIriGetV1()` returns a ZIO of Option, so you could even call on that one `getOrFailWith`:

```scala
result <- 
  userProfileByIRIGetV1(userIri, profileType)
    .getOrFailWith(NotFoundException(s"User '$userIRI' not found"))
    .map(UserProfileResponseV1(_))
```

Reply:
Somehow `getOrFailWith` on your snipped can't be applied, but I make it a bit better with `ZIO.fromOption`.